pkg/agent/cniserver/server.go

```diff
@@ -442,6 +442,10 @@ func (s *CNIServer) CmdAdd(ctx context.Context, request *cnipb.CniCmdRequest) (*
 
 	result := &ipam.IPAMResult{Result: current.Result{CNIVersion: current.ImplementedSpecVersion}}
 	netNS := s.hostNetNsPath(cniConfig.Netns)
+	if !validateRuntime(netNS) {
+		// Support for Docker runtime on Windows has been removed since Antrea 2.0.
+		return nil, fmt.Errorf("failed to process CmdAdd request because Docker runtime is not supported after Antrea 2.0")
+	}
```

Comment:
nit: this may be more elegant:

```go
if err := validateRuntime(netNS); err != nil {
        return nil, fmt.Errorf("failed to validate container runtime for CmdAdd request: %w")
}
```

with (for Windows):
```go
func validateRuntime(netNS string) bool {
	if isDockerContainer(netNS) {
		return fmt.Errorf("Docker runtime is not supported after Antrea 2.0 for Windows Nodes")
	}
	return nil
}
```

This way we don't have to add any Windows-specific comment to server.go.

I would say the code is self-explanatory in this case, and no extra comment is needed.

Reply:
Updated, thanks.